pkg/virt-controller/services/template.go

```diff
@@ -372,14 +372,7 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 		return i.Slirp != nil
 	})
 	if len(slirpIfaces) > 0 {
-		if plugin := ReadNetBindingPluginConfiguration(t.clusterConfig.GetConfig(), SlirpNetworkBindingPluginName); plugin == nil {
-			return nil, fmt.Errorf("couldn't find %s network binding plugin configuration, make sure its specified in Kubevirt config", SlirpNetworkBindingPluginName)
-		} else {
-			requestedHookSidecarList = append(requestedHookSidecarList, hooks.HookSidecar{
-				Image:           plugin.SidecarImage,
-				ImagePullPolicy: t.clusterConfig.GetImagePullPolicy(),
-			})
-		}
+		requestedHookSidecarList = append(requestedHookSidecarList, SlirpNetBindingPluginSidecar(vmi, t.clusterConfig.GetConfig(), t.recorder))
```

Comment:
Can you please move all the slirp related code to `NetBindingPluginSidecarList`? So the template file won't have any of this logic and all the binding sidecar logic will reside in the net_binding file.

Reply:
Note that `SlirpNetBindingPluginSidecar` now requires the event recorder, it will require changing `NetBindingPluginSidecarList ` to receive it, WDYT?

Reply:
Done